v2/internal/testcommon/direct_client.go

```diff
@@ -20,6 +20,8 @@ type directClient struct {
 	inner client.Client
 }
 
+var _ client.Client = &directClient{}
```

Comment:
This assertion already exists on line 105 -- maybe just move it up here rather than duplicating it?

Reply:
I didn't spot it at the end of the file. I've tidied up in favour of the one near the top.